src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -634,7 +634,8 @@ public boolean verifyEvidence(Evidence evidence) throws RealmUnavailableExceptio
                 mechanismRealmConfiguration = MechanismRealmConfiguration.NO_REALM;
             }
         } else {
-            return checkEvidenceTrusted(evidence) ? true : stateRef.get().verifyEvidence(evidence);
+            final boolean verified = stateRef.get().verifyEvidence(evidence);
+            return (evidence instanceof SecurityIdentityEvidence) ? verified && authorize() : verified;
```

Comment:
This could be simplified e.g.: return verified && (! (evidence instanceof SecurityIdentityEvidence) || authorize());